src/test/regress/sql/role_operations_from_non_maindb.sql

```diff
@@ -0,0 +1,106 @@
+-- Create a new database
+set citus.enable_create_database_propagation to on;
+CREATE DATABASE role_operations_test_db;
+SET citus.superuser TO 'postgres';
+-- Connect to the new database
+\c role_operations_test_db
+-- Test CREATE ROLE with various options
+CREATE ROLE test_role1 WITH LOGIN PASSWORD 'password1';
+
+\c role_operations_test_db - - :worker_1_port
+CREATE USER "test_role2-needs\!escape"
+WITH
+    SUPERUSER CREATEDB CREATEROLE INHERIT LOGIN REPLICATION BYPASSRLS CONNECTION
+LIMIT 10 VALID UNTIL '2023-01-01' IN ROLE test_role1;
+
+\c regression - - :master_port
+
+select result FROM run_command_on_all_nodes($$
+    SELECT array_to_json(array_agg(row_to_json(t)))
+    FROM (
+        SELECT rolname, rolsuper, rolinherit, rolcreaterole, rolcreatedb,
+        rolcanlogin, rolreplication, rolbypassrls, rolconnlimit,
+        (rolpassword != '') as pass_not_empty, DATE(rolvaliduntil)
+        FROM pg_authid
+        WHERE rolname in ('test_role1', 'test_role2-needs\!escape')
+        ORDER BY rolname
+    ) t
+$$);
+
+select result FROM run_command_on_all_nodes($$
+    SELECT array_to_json(array_agg(row_to_json(t)))
+    FROM (
+        SELECT r.rolname
+        FROM pg_dist_object d
+        JOIN pg_roles r ON d.objid = r.oid
+        WHERE r.rolname IN ('test_role1', 'test_role2-needs\!escape')
+        order by r.rolname
+    ) t
+$$);
+
+\c role_operations_test_db - - :master_port
+-- Test ALTER ROLE with various options
+ALTER ROLE test_role1 WITH PASSWORD 'new_password1';
+
+\c role_operations_test_db - - :worker_1_port
+ALTER USER "test_role2-needs\!escape"
+WITH
+    NOSUPERUSER NOCREATEDB NOCREATEROLE NOINHERIT NOLOGIN NOREPLICATION NOBYPASSRLS CONNECTION
+LIMIT 5 VALID UNTIL '2024-01-01';
+
+\c regression - - :master_port
+select result FROM run_command_on_all_nodes($$
+    SELECT array_to_json(array_agg(row_to_json(t)))
+    FROM (
+        SELECT rolname, rolsuper, rolinherit, rolcreaterole, rolcreatedb,
+        rolcanlogin, rolreplication, rolbypassrls, rolconnlimit,
+        (rolpassword != '') as pass_not_empty, DATE(rolvaliduntil)
+        FROM pg_authid
+        WHERE rolname in ('test_role1', 'test_role2-needs\!escape')
+        ORDER BY rolname
+    ) t
+$$);
+
+\c role_operations_test_db - - :master_port
+-- Test DROP ROLE
+DROP ROLE IF EXISTS test_role1, "test_role2-needs\!escape";
+
+
```

Comment:
please avoid adding trailing new lines because they make the code ugly a lot

```suggestion
```

Reply:
Done